web/app.js

```diff
@@ -2061,6 +2063,22 @@ const PDFViewerApplication = {
     return wholeTicks;
   },
 
+  _accumulateFactor(previousScale, factor, prop) {
+    if (factor === 1) {
+      return 1;
+    }
+    // If the direction changed, reset the accumulated ticks.
+    if ((this[prop] > 1 && factor < 1) || (this[prop] < 1 && factor > 1)) {
+      this[prop] = 1;
+    }
+
+    const newFactor = previousScale * factor * this[prop];
+    const wholeFactor = Math.floor(newFactor * 100) / (100 * previousScale);
+    this[prop] = factor / wholeFactor;
+
+    return wholeFactor;
```

Comment:
The `wholeFactor` variable name feels slightly misleading, since it doesn't normally contain a whole number in this method (as opposed to the method above, from which I guess this was copied). How about we simply do the following?
```suggestion
    const newFactor =
      Math.floor(100 * previousScale * factor * this[prop]) / (100 * previousScale);
    this[prop] = factor / newFactor;
    return newFactor;
```